src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -0,0 +1,543 @@
+
+Non-symmetric path-conservative two-point flux discretizing the nonconservative (source) term
+that contains the gradients of the bottom topography and waterheights from the coupling between layers
+and the nonconservative pressure formulation [`ShallowWaterMultiLayerEquations1D`](@ref).
+
+When the bottom topography is nonzero this scheme will be well-balanced when used with [`flux_ersing_etal`](@ref).
+
+In the two-layer setting this combination is equivalent to the fluxes in:
+- Patrick Ersing, Andrew R. Winters (2023)
+  An entropy stable discontinuous Galerkin method for the two-layer shallow water equations on 
+  curvilinear meshes
+  [DOI: 10.48550/arXiv.2306.12699](https://doi.org/10.48550/arXiv.2306.12699)
+"""
+@inline function Trixi.flux_nonconservative_ersing_etal(u_ll, u_rr,
+                                                        orientation::Integer,
+                                                        equations::ShallowWaterMultiLayerEquations1D)
+    # Pull the necessary left and right state information
+    h_ll = waterheight(u_ll, equations)
+    h_rr = waterheight(u_rr, equations)
+    b_rr = u_rr[end]
+    b_ll = u_ll[end]
+
+    # Compute the jumps
+    h_jump = h_rr - h_ll
+    b_jump = b_rr - b_ll
+    g = equations.gravity
+
+    # Initialize flux vector
+    f = zero(MVector{2 * nlayers(equations) + 1, real(equations)})
+
+    # Compute the nonconservative flux in each layer (0, ..., 0, f_hv[1], ..., f_hv[NLAYERS], 0)
+    # with f_hv[i] = gh[i] * (b + ∑h[k] + ∑σ[k]h[k])_x.  
+    for i in eachlayer(equations)
+        f_hv = g * h_ll[i] * b_jump
+        for j in eachlayer(equations)
+            if j < i
+                f_hv += g * h_ll[i] *
+                        (equations.rhos[j] / equations.rhos[i] * h_jump[j])
+            else # (i<j<nlayers) nonconservative formulation of the pressure
+                f_hv += g * h_ll[i] * h_jump[j]
+            end
+        end
+        setindex!(f, f_hv, i + nlayers(equations))
+    end
+
+    return SVector(f)
+end
+
+"""
+    flux_ersing_etal(u_ll, u_rr, orientation::Integer,
+                                     equations::ShallowWaterMultiLayerEquations1D)
+
+Total energy conservative (mathematical entropy for MLSWE) split form,
+without the hydrostatic pressure.
+When the bottom topography is nonzero this scheme will be well-balanced when used with the 
+nonconservative [`flux_nonconservative_ersing_etal`](@ref).
+
+To obtain an entropy stable formulation the `surface_flux` can be set as
+[`FluxPlusDissipation(flux_ersing_etal, DissipationLocalLaxFriedrichs()), flux_nonconservative_ersing_etal`](@ref).
+
+In the two-layer setting this combination is equivalent to the fluxes in:
+- Patrick Ersing, Andrew R. Winters (2023)
+  An entropy stable discontinuous Galerkin method for the two-layer shallow water equations on 
+  curvilinear meshes
+  [DOI: 10.48550/arXiv.2306.12699](https://doi.org/10.48550/arXiv.2306.12699)
```

Comment:
```suggestion
  [DOI: 10.1007/s10915-024-02451-2](https://doi.org/10.1007/s10915-024-02451-2)
```
same here

Reply:
Done in https://github.com/trixi-framework/TrixiShallowWater.jl/commit/53dfc00487c4527dbad287db70ea222fb6459e43